jena-arq/src/main/java/org/apache/jena/sparql/core/mem/PMapQuadTable.java

```diff
@@ -37,13 +39,29 @@
  * use.
  *
  */
-public abstract class PMapQuadTable extends PMapTupleTable<FourTupleMap, Quad>implements QuadTable {
+public class PMapQuadTable extends PMapTupleTable<FourTupleMap, Quad, Consumer4<Node>>implements QuadTable {
+
+
+    private static class QuadTupleMap extends TupleMap {
+
+        private QuadTupleMap(final String mappedOrder) {
+            super(mappedOrder, compileMapping("GSPO", mappedOrder));
+        }
+    }
+
```

Comment:
It seems significantly clearer to me this way (the `static` methods just delegate to constructors, and it's clear from the outside that they must) but I have no problem doing it the other way. I'll add a commit "resetting" `TupleMap` and using `static`s later today.